core/src/sumeragi/network_topology.rs

```diff
@@ -168,38 +164,31 @@ impl Builder {
         self
     }
 
-    /// Set number of view changes after the latest committed block. Default: 0
-    pub fn with_view_changes(mut self, view_change_proofs: ViewChangeProofs) -> Self {
-        self.view_change_proofs = view_change_proofs;
-        self
-    }
-
     /// Build and get topology.
     ///
     /// # Errors
     /// 1. Required field is omitted.
     /// 2. No peer exists.
-    pub fn build(self) -> Result<Topology> {
+    pub fn build(self, number_of_view_changes_this_round: u64) -> Result<Topology> {
         let peers = field_is_some_or_err!(self.peers)?;
         if peers.is_empty() {
             return Err(eyre!("There must be at least one peer in the network."));
         }
         let at_block = field_is_some_or_err!(self.at_block)?;
         let peers: Vec<_> = peers.into_iter().collect();
-        let n_view_changes = self.view_change_proofs.len();
-        let since_last_shuffle = n_view_changes % peers.len();
+        let n_view_changes = number_of_view_changes_this_round;
+        let since_last_shuffle = n_view_changes % (peers.len() as u64);
         let is_full_circle = since_last_shuffle == 0;
         let sorted_peers = if is_full_circle {
-            sort_peers_by_hash_and_counter(peers, &at_block, n_view_changes as u64)
+            sort_peers_by_hash_and_counter(peers, &at_block, n_view_changes)
```

Comment:
Small note on `sort_peers_by_hash_and_coundter`:
- allocation could be removed
- sorting seems to be redundant and degrade performance (or what purpose of sorting here?)
- name probably should be `shuffle_peers_by_hash_and_counter`